components/movementsensor/imuvectornav/imu.go

```diff
@@ -183,42 +183,32 @@ func newVectorNav(
 	accVpeTunning = append(accVpeTunning, utils.BytesFromFloat32LE(10)...)
 	accVpeTunning = append(accVpeTunning, utils.BytesFromFloat32LE(10)...)
 
-	err = v.writeRegisterSPI(ctx, vpeAccTunning, accVpeTunning)
-	if err != nil {
+	if err = v.writeRegisterSPI(ctx, vpeAccTunning, accVpeTunning); err != nil {
 		return nil, errors.Wrap(err, "couldn't set vpe adaptive tunning")
 	}
-	err = v.writeRegisterSPI(ctx, deltaVDeltaThetaConfig, []byte{0, 0, 0, 0, 0, 0})
-	if err != nil {
+	if err = v.writeRegisterSPI(ctx, deltaVDeltaThetaConfig, []byte{0, 0, 0, 0, 0, 0}); err != nil {
 		return nil, errors.Wrap(err, "couldn't configure deltaV register")
 	}
+
 	// tare the heading
-	err = v.vectornavTareSPI(ctx)
-	if err != nil {
+	if err = v.vectornavTareSPI(ctx); err != nil {
 		return nil, err
 	}
-	smp := uint(100)
-	if v.polling > 0 {
-		smp = uint(v.polling)
-	}
-
 	// compensate for acceleration bias due to misalignement
-	err = v.compensateAccelBias(ctx, smp)
-	if err != nil {
+	if err = v.compensateAccelBias(ctx, pollFreq); err != nil {
 		return nil, err
 	}
 	// compensate for constant DV bias in mesurament
-	err = v.compensateDVBias(ctx, smp)
-	if err != nil {
+	if err = v.compensateDVBias(ctx, pollFreq); err != nil {
 		return nil, err
 	}
 
-	// optionally start a polling goroutine
-	if pfreq > 0 {
-		logger.CDebugf(ctx, "vecnav: will pool at %d Hz", pfreq)
+	if pollFreq > 0 {
```

Comment:
polling frequency is alway greater than zero because we set it to 200 at line 126 or conf value